src/software/utils/main_fisheyeProjection.cpp

```diff
@@ -0,0 +1,431 @@
+ * @param[in] iter - index of input image for merging into panorama
+ * @param[in] rotations - contains adjustment rotations on each image set by user
+ * @param[out] imageOut - output panorama which is incremented at each call
+ */
+void fisheyeToEquirectangular(image::Image<image::RGBAfColor>& imageIn, const int nbImages, int iter, const std::array<std::vector<double>, 3> rotations, image::Image<image::RGBAfColor>& imageOut)
+{
+  std::size_t inWidth = imageIn.Width();
+  std::size_t inHeight = imageIn.Height();
+  std::size_t inSize = std::min(inWidth, inHeight);
+
+  double zRotation = double(30.0 - iter * 360.0 / nbImages) + rotations[2].at(iter);
+  double xRotation = double(-abs(90.0 - abs(zRotation))/30.0) + rotations[0].at(iter);
+  double yRotation = rotations[1].at(iter);
+
+  const image::Sampler2d<image::SamplerLinear> sampler;
+  for(int j = 0; j < inSize; ++j)
+  {
+    for(int i = 0; i < 2 * inSize; ++i)
+    {
+      image::RGBAfColor pixel(0.f, 0.f, 0.f, 0.f);
+
+      Vec3 ray = SphericalMapping::get3DPoint(Vec2(i,j), 2*inSize, inSize);
+
+      ray = rotationXYZ(degreeToRadian(xRotation), degreeToRadian(yRotation), degreeToRadian(zRotation)) * ray;
+      const Vec2 x = SphericalMapping::get2DCoordinates(ray, inSize);
+      const Vec2 xFish(inWidth/2 - x(0), inHeight/2 - x(1));
+
+      pixel = sampler(imageIn, xFish(1), xFish(0));
+      float alpha = pixel.a();
+
+      imageOut(j, i).r() = pixel.r()*alpha + imageOut(j, i).r()*(1.f-alpha);
+      imageOut(j, i).g() = pixel.g()*alpha + imageOut(j, i).g()*(1.f-alpha);
+      imageOut(j, i).b() = pixel.b()*alpha + imageOut(j, i).b()*(1.f-alpha);
+      imageOut(j, i).a() = pixel.a()*alpha + imageOut(j, i).a()*(1.f-alpha);
+    }
+  }
+}
+
+
+/**
+ * @brief Load input images and call functions to stitch 360° panorama
+ * @param[in] imagePaths - input images paths
+ * @param[in] metadatas - input metadata for each image
+ * @param[in] rotations - contains adjustment rotations on each image set by user
+ * @param[out] outputFolder - output folder path to write panorama
+ */
+void stitchPanorama(const std::vector<std::string>& imagePaths, const std::vector<oiio::ParamValueList>& metadatas, const float blurWidth, const std::array<std::vector<double>, 3> rotations, std::string& outputPath)
+{
+  int nbImages = imagePaths.size();
+  image::Image<image::RGBAfColor> imageOut;
+  std::vector<oiio::ImageBuf> buffers(nbImages);
+  oiio::ImageBuf& bufferOut = buffers[0];
+  std::size_t inSize;
+
+  for(int i=0; i<nbImages; ++i)
+  {
+    oiio::ImageBuf& buffer = buffers[i];
+    image::Image<image::RGBfColor> imageIn;
+    image::Image<image::RGBAfColor> imageAlpha;
+
+    ALICEVISION_LOG_INFO("Projecting " << imagePaths[i] << " into equirectangular space");
+
+    try
+    {
+      image::readImage(imagePaths[i], imageIn, image::EImageColorSpace::LINEAR);
+    }
+    catch(fs::filesystem_error& e)
+    {
+        ALICEVISION_LOG_ERROR("Can't open image file : " << imagePaths[i]);
```

Comment:
shouldn't we also exit or do something? if we catch then we go on and `imageIn` is used